x/gamm/keeper/migrate.go

```diff
@@ -13,56 +13,44 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
-func (k Keeper) Migrate(ctx sdk.Context, sender sdk.AccAddress, sharesToMigrate sdk.Coin, poolIdEntering uint64) (amount0, amount1 sdk.Int, liquidity sdk.Dec, poolIdLeaving uint64, err error) {
+// Migrate migrates unlocked lp tokens from a balancer pool to a concentrated liquidity pool.
+// Fails if the lp tokens are locked (must utilize UnlockAndMigrate function in the superfluid module)
+func (k Keeper) Migrate(ctx sdk.Context, sender sdk.AccAddress, sharesToMigrate sdk.Coin) (amount0, amount1 sdk.Int, liquidity sdk.Dec, poolIdLeaving, poolIdEntering uint64, err error) {
 	// Get the balancer poolId by parsing the gamm share denom.
-	poolIdLeaving, err = getPoolIdFromSharesDenom(sharesToMigrate.Denom)
-	if err != nil {
-		return sdk.Int{}, sdk.Int{}, sdk.Dec{}, 0, err
-	}
+	poolIdLeaving = types.MustGetPoolIdFromShareDenom(sharesToMigrate.Denom)
```

Comment:
Do you think I should reintroduce the function I removed from this PR? I basically wanted to utilize a function that already existed but I realize here that we panic if the denom does not exist. Is this problematic? If so, I will reintroduce the function I had made previously for this.

Reply:
Maybe it's a open question, but I personally see that unifying error / panic as much as possible, and letting the caller decide what to do with the error as a better design, so that we don't have multiple places that errors or panics.(Setters and getters from state are unevitable for panicing). But again, its an open ended question, I'll leave it up to you to decide on this :)

Reply:
I personally think unifying panics and errors as much as possible is a better design so that we don't have multiple places here and there that either errors or panics, so that the caller of this function can have a unified point of handling errors (the must getters and must setters from state is inevitable) . But again, I guess this is an open ended question, I'll leave it to you to decide :)

Reply:
Agreed! Addressed here https://github.com/osmosis-labs/osmosis/pull/4129/commits/d3bdaa11aa1768044f843bd148388a498601c318